### PR TITLE
Update draft -- Sriram's edits

### PR DESCRIPTION
**Pull request by ksriram25**
Went through Jeff's changes carefully and did some of my own edits.
Added new wording to clarify the text in some places.
Fixed an error in example B.2. Tweaked it a bit.

Alvaro's comment at the mike (IETF 115)? What wording if any do we want to add?

### CODE REVIEW

File: draft-ietf-idr-deprecate-as-set-confed-set.xml [hunk 1]
```diff
@@ -233,21 +209,22 @@ Route aggregation that was previously performed by proxy aggregation without the
...
-      AS_SETs, conformant BGP speakers SHOULD use the "Treat-as-withdraw" error
+      AS_SETs, conformant BGP speakers SHOULD use the "treat-as-withdraw" error
       handling behavior as per <xref target="RFC7606"/>.</t>
 
       <section title="BGP AS_PATH &quot;Brief&quot; Aggregation">
-        <t><xref target="RFC4271"/>, Sections 9.1.4 and 9.2.2.2, describes BGP
-        aggregation procedures.  <xref target="RFC4271"/>, Appendix F.6
+        <t> 
+Sections 9.1.4 and 9.2.2.2 of <xref target="RFC4271"/> describe BGP aggregation procedures. 
+Appendix F.6 in <xref target="RFC4271"/>
```
Comment (jhaas-pfrc) [inline draft-ietf-idr-deprecate-as-set-confed-set.xml]: Please preserve indentation.

File: draft-ietf-idr-deprecate-as-set-confed-set.xml [hunk 2]
```diff
@@ -307,11 +287,17 @@ Route aggregation that was previously performed by proxy aggregation without the
         <t>In order to ensure a consistent BGP origin AS is announced for
         aggregate BGP routes for implementations of "brief" BGP aggregation, the
         implementation should be configured to truncate the AS_PATH after the
-        right-most instance of the desired origin AS for the aggregate.</t>
+        right-most instance of the desired origin AS for the aggregate. The desired origin AS could be the aggregator AS itself.</t>
```
Comment (jhaas-pfrc) [inline draft-ietf-idr-deprecate-as-set-confed-set.xml]: Wrap line.

File: draft-ietf-idr-deprecate-as-set-confed-set.xml [hunk 3]
```diff
@@ -307,11 +287,17 @@ Route aggregation that was previously performed by proxy aggregation without the
...
-        right-most instance of the desired origin AS for the aggregate.</t>
+        right-most instance of the desired origin AS for the aggregate. The desired origin AS could be the aggregator AS itself.</t>
 
         <t>If the resulting AS_PATH would be truncated from the otherwise
-        expected result of BGP AS_PATH aggregation (an AS_SET would be
-        generated, or ASes are removed from the "longest leading sequence" of
+        expected result of BGP AS_PATH aggregation (an AS_SET would not be
+        generated, and/or ASes are removed from the "longest leading sequence" of
+
+<!--
```
Comment (jhaas-pfrc) [inline draft-ietf-idr-deprecate-as-set-confed-set.xml]: The change above seems right.  Remove comment.

File: draft-ietf-idr-deprecate-as-set-confed-set.xml [hunk 4]
```diff
@@ -322,7 +308,10 @@ Route aggregation that was previously performed by proxy aggregation without the
       <t>When aggregating prefixes, network operators MUST use brief
       aggregation. In brief aggregation, the AGGREGATOR and ATOMIC_AGGREGATE
       Path Attributes are included, but the AS_PATH does not have AS_SET or
-      AS_CONFED_SET path segment types.</t>
+      AS_CONFED_SET path segment types. 
+See <xref target="brief-agg-example"/> for examples of 
+brief aggregation while keeping the origin AS unambiguous 
+and generating appropriate ROAs.</t>
```
Comment (jhaas-pfrc) [inline draft-ietf-idr-deprecate-as-set-confed-set.xml]: Fix indentation.

File: draft-ietf-idr-deprecate-as-set-confed-set.xml [hunk 5]
```diff
@@ -334,9 +323,10 @@ Route aggregation that was previously performed by proxy aggregation without the
       this filtering policy.</t>
 
       <t>Operators MUST install egress filters to block data packets when the
-      destination address belongs to an internal prefix. Similarly, any known
+      destination address belongs to an internal prefix.
+	Similarly, any known
```
Comment (jhaas-pfrc) [inline draft-ietf-idr-deprecate-as-set-confed-set.xml]: Not sure why the line split here.

File: draft-ietf-idr-deprecate-as-set-confed-set.xml [hunk 6]
```diff
@@ -420,10 +409,14 @@ Route aggregation that was previously performed by proxy aggregation without the
...
-      then p/22 will not be announced to AS1, AS2, AS3, or AS4. Instead, as
-      further illustration, p1/24, p2/24 and p4/24 are announced to AS3.</t>
+
+   <t> 
+Presented here is an illustration of how an AS_SET is not used when aggregating and still data-plane route loops are avoided. 
+Consider that p1/24 (from AS 64501), p2/24 (from AS 64502), p3/24 (from AS 64503), and p4/24 (from AS 64504) are aggregated by AS 64505 to p/22.
+AS_SET is not used with the aggregate p/22 but AGGREGATOR and ATOMIC AGGREGATE are used. 
+Data-plane route loops are avoided by not announcing the aggregate p/22 to the contributing ASes, i.e, AS 64501, AS 64502, AS 64503, and AS 64504.  
+Instead, as further illustration, p1/24, p2/24, and p4/24 are announced to AS 64503. The routing tables (post aggregation) of each of the ASes are depicted in the diagram below .
+</t>
```
Comment (jhaas-pfrc) [inline draft-ietf-idr-deprecate-as-set-confed-set.xml]: Fix indentation.

File: draft-ietf-idr-deprecate-as-set-confed-set.xml [hunk 7]
```diff
@@ -468,10 +461,14 @@ p4/24 AS_PATH "64504"
       </artset>
     </section>
     <section title="Examples of Inconsistent BGP Origin-AS Generated by
-                    Traditional Brief-Style Aggregation">
+                    Traditional Brief Aggregation" anchor="brief-agg-example">
+<t>
+In the examples below, it is illustrated how brief aggregation may result in inconsistent origin AS.   
+</t>
+
```
Comment (jhaas-pfrc) [inline draft-ietf-idr-deprecate-as-set-confed-set.xml]: Fix indentation and line wrap.

File: draft-ietf-idr-deprecate-as-set-confed-set.xml [hunk 8]
```diff
@@ -504,9 +501,19 @@ R4 - 192.0.2.192/26 AS_PATH "64504 64501"
...
+<!-- 
+XXX KS: I make a correction above from "64504 [ 64501 ]" 
+to "[ 64504 64501 ] because "[ 64504 ] 64501" does not follow
+from algorithm in RFC 4271. 64504 is not common to both
+contributing paths. So it must also be included in the AS_SET.
+-->    
+
+<!-- XXX KS: This is replaced (as below):  <t>If brief aggregation is in use, 
+the AS_PATH is truncated to "64504".</t>  -->
```
Comment (jhaas-pfrc) [inline draft-ietf-idr-deprecate-as-set-confed-set.xml:523]: The error here you're noting is valid.
The intent is to have an example with a common leading AS with trailing AS that is thrown into a set that is discarded.  You can either update this in your diff, or leave it for me to update in mine once we merge your stuff in.

File: draft-ietf-idr-deprecate-as-set-confed-set.xml [hunk 9]
```diff
@@ -526,7 +533,7 @@ R4 - 192.0.2.192/26 AS_PATH "64504 64501"
 
       <section title="Achieving Consistent Origin-AS During Aggregation">
         <t>In the three scenarios above, the aggregating AS 64500 is using
-        traditional brief style aggregation.  This results in inconsistent
+        traditional brief aggregation.  This results in inconsistent
```
Comment (jhaas-pfrc) [inline draft-ietf-idr-deprecate-as-set-confed-set.xml:545]: brief-style?

File: draft-ietf-idr-deprecate-as-set-confed-set.xml [hunk 10]
```diff
@@ -544,10 +551,12 @@ R4 - 192.0.2.192/26 AS_PATH "64504 64501"
         need to be created for each of the more specific prefixes.</t>
 
         <t>In more complex proxy aggregation scenarios, there may be a desire
-        to permit some stable portion of the contributing AS to be kept in the
+        to permit some stable (i.e., common) portion of the contributing AS_PATHs to be kept in the
         aggregate route.  Consider the case for Scenario 3, where the neighbor
         AS is the same for both R3 and R4 - AS 64504.  In such a case, an
-        implementation may permit the aggregate's brief AS_PATH to be "64504".</t>
+        implementation may permit the aggregate's brief AS_PATH to be "64504", and
+a ROA would be created for the aggregate prefix with 64504 as the origin AS.
```
Comment (jhaas-pfrc) [inline draft-ietf-idr-deprecate-as-set-confed-set.xml]: indentation.

File: draft-ietf-idr-deprecate-as-set-confed-set.xml [hunk 11]
```diff
@@ -268,10 +245,13 @@ Route aggregation that was previously performed by proxy aggregation without the
...
         </blockquote>
 
-        <t>When BGP AS_PATH aggregation is done according to the Section 9.2.2.2
+        <t>When BGP AS_PATH aggregation is done according to the <xref target="RFC4271" section="9.2.2.2" sectionFormat="comma"/>,
         procedures and any resulting AS_SETs are discarded, this is typically
-        referred to as "brief" aggreation in implementations.  This results in
-        an AS_PATH that has the property (from Section 9.2.2.2):</t>
+        referred to as "brief" aggregation in implementations.
+(Note: It is true in practice and assumed here that AS_SETs 
+occur only at the beginning (origination side) of the AS_PATHs.)
```
Comment (jhaas-pfrc) [inline draft-ietf-idr-deprecate-as-set-confed-set.xml]: I would skip this parenthetical (and mis-indented) statement.  It doesn't necessarily help with the explanation, and the "beginning" portion isn't how we'd normally refer to this in RFC land.

File: draft-ietf-idr-deprecate-as-set-confed-set.xml [hunk 12]
```diff
@@ -268,10 +245,13 @@ Route aggregation that was previously performed by proxy aggregation without the
...
+Brief aggregation results in an AS_PATH that has the property 
+(from <xref target="RFC4271" section="9.2.2.2" sectionFormat="comma"/>):</t>
```
Comment (jhaas-pfrc) [inline draft-ietf-idr-deprecate-as-set-confed-set.xml]: Fix indentation, keep this.

File: draft-ietf-idr-deprecate-as-set-confed-set.xml (hunk 1 above)
Comment (ksriram25) [reply draft-ietf-idr-deprecate-as-set-confed-set.xml]: OK, sure.

File: draft-ietf-idr-deprecate-as-set-confed-set.xml (hunk 8 above)
Comment (ksriram25) [reply draft-ietf-idr-deprecate-as-set-confed-set.xml:523]: Yes, your origin example B.2 didn't follow this condition from rfc 4271:

           - all tuples of type AS_SEQUENCE in the aggregated AS_PATH
             SHALL appear in all of the AS_PATHs in the initial set of
             routes to be aggregated. 

As I see it B.3 is already "an example with a common leading AS with trailing AS that is thrown into a set that is discarded."
I leave this up to you to sort out.

File: draft-ietf-idr-deprecate-as-set-confed-set.xml (hunk 9 above)
Comment (ksriram25) [reply draft-ietf-idr-deprecate-as-set-confed-set.xml:545]: You were varyingly using "brief aggregation", "brief-style aggregation" and "brief style aggregation" but all meant the same. I tried to retain only one form "brief aggregation". Eliminated the attachment of 'style'.

File: draft-ietf-idr-deprecate-as-set-confed-set.xml (hunk 10 above)
Comment (ksriram25) [reply draft-ietf-idr-deprecate-as-set-confed-set.xml]: done

File: draft-ietf-idr-deprecate-as-set-confed-set.xml (hunk 11 above)
Comment (ksriram25) [reply draft-ietf-idr-deprecate-as-set-confed-set.xml]: done

File: draft-ietf-idr-deprecate-as-set-confed-set.xml (hunk 12 above)
Comment (ksriram25) [reply draft-ietf-idr-deprecate-as-set-confed-set.xml]: done

File: draft-ietf-idr-deprecate-as-set-confed-set.xml (hunk 7 above)
Comment (ksriram25) [reply draft-ietf-idr-deprecate-as-set-confed-set.xml]: done

File: draft-ietf-idr-deprecate-as-set-confed-set.xml (hunk 6 above)
Comment (ksriram25) [reply draft-ietf-idr-deprecate-as-set-confed-set.xml]: done

File: draft-ietf-idr-deprecate-as-set-confed-set.xml (hunk 5 above)
Comment (ksriram25) [reply draft-ietf-idr-deprecate-as-set-confed-set.xml]: fixed

File: draft-ietf-idr-deprecate-as-set-confed-set.xml (hunk 2 above)
Comment (ksriram25) [reply draft-ietf-idr-deprecate-as-set-confed-set.xml]: fixed

File: draft-ietf-idr-deprecate-as-set-confed-set.xml (hunk 3 above)
Comment (ksriram25) [reply draft-ietf-idr-deprecate-as-set-confed-set.xml]: done

File: draft-ietf-idr-deprecate-as-set-confed-set.xml (hunk 4 above)
Comment (ksriram25) [reply draft-ietf-idr-deprecate-as-set-confed-set.xml]: done

**Review by jhaas-pfrc — APPROVED** [review]
Again, approve, merge and sort out the lingering issues in -10 branch.

**Pull request merged**